_posts/2022-12-16-month-of-mlops-third-prize-cheque-easy.md

```diff
@@ -0,0 +1,193 @@
+
+
+### Deployment Trigger
+After training and evaluating the performance of our model against a test dataset, we check if the model accuracy is above a desired threshold. Only if the model satisfies this criteria, it will be deployed to a REST API endpoint.
+
+
+### Model Deployer
+Once the deployment trigger has been activated, this step takes the trained model and deploys it using ZenML's Mlflow Deployer and creates a prediction service corresponding to the deployed model.
+
+
+## Inference Pipeline
+This pipeline consists of 3 simple steps as summarized below:
+
+![inference_pipeline](../assets/posts/mlops-comp-2022-cheque-easy/inference_pipeline_cheque_easy.png)
+
+### Load data
+Import the data (i.e. cheque image) that will be fed as input to model for getting prediction.
+
+
+### Load Prediction Service
+This step was used to load the MLFlow model prediction service.
+
+
+### Predict
+Finally, the input image will be encoded as a base64 string and fed as input to the MLFlow model prediction service.  This will return the json output generated by the model containing the extracted cheque details.
+
+
+Thanks to ZenML, I was easily able to build reproducible and maintainable pipelines with including support for annotation, experiment tracking, model deployment, pipeline orchestration, secret management, artfifact storage, etc.
+
+
+## Gradio Demo
+Finally, I created a simple [demo](https://huggingface.co/spaces/shivi/ChequeEasy) of the project to interact with the trained model using gradio and deployed it with help of Hugging Face Spaces.
+
+
+## Future Work
+There's definitely a lot of scope in terms of improving the performance of the Donut model so that it can work well with different kinds of cheque image inputs.
+Since this competition was more about MLOps, I built a basic working model that would suffice for solving the use case and spent more effort on the MLOps side of the solution.
+Going forward, I would definitely like to focus on the modeling aspects of this solution. I would even like to test out Donut's multi-lingual capabilities to process cheque images written in languages other than English.
+
+On the MLOps side of the solution, I would like to explore ZenML's DeepChecks integration for data and model validation to further enhance my solution.
+
+## 💭 Conclusions
+As an individual, I had so much fun participating in this competition. It turned out to be a great opportunity to show case my work to top most MLOps industry experts, build my own small open source project, learn from the ZenML community as well as contribute back to ZenML. It's hard to get so much out of just one competition and it was only possible because of how well the ZenML team organized it.
```

Comment:
```suggestion
As an individual, I had so much fun participating in this competition. It turned out to be a great opportunity to showcase my work to the topmost MLOps industry experts, build my own small open-source project, learn from the ZenML community as well as contribute back to ZenML. It's hard to get so much out of just one competition and it was only possible because of how well the ZenML team organized it.
```